tests/integration/Owncloud/OcisPhpSdk/DriveTest.php

```diff
@@ -72,15 +72,13 @@ public function testDeleteEnabledDrive(): void
 
     public function testGetDriveRole(): void
     {
-        // At the time of writing, "stable" is major version 5 of ocis.
-        // This functionality works with major version 6.
-        // When ocis major version 6 has been released as "stable" then remove this test skip.
-        if (getenv('OCIS_VERSION') === "stable") {
+        try {
+            $role = $this->drive->getRoles();
+        } catch(\Exception) {
```

Comment:
```suggestion
        } catch(EndPointNotImplementedException) {
```

We can just catch the exception that we think could happen.
Then in the "catch" block,
`if (getenv('OCIS_VERSION') !== "stable")`
then fail the test.
Something like:
`$this->fail("EndPointNotImplementedException was thrown unexpectedly");`

And the `assertContainsOnlyInstancesOf` code will need to be put inside the `try` block - when we do get the `EndPointNotImplementedException` then we do not want to check the content of `$roles`

This would be a way to implement both test flows, for OCIS 5 and 6, within one test case.

Otherwise, write 2 completely separate test cases, and do the appropriate skipping so that only one test cazse runs in each pipeline.